src/app/centroidController.js

```diff
@@ -0,0 +1,263 @@
+define([
+  'app/config',
+  'app/graphicsUtils',
+
+  'dojo/_base/lang',
+  'dojo/Deferred',
+  'dojo/promise/all',
+  'dojo/topic',
+  'dojo/text!app/templates/projectPopupTemplate.html',
+
+  'dijit/TooltipDialog',
+  'dijit/popup',
+
+  'esri/layers/FeatureLayer',
+  'esri/tasks/query'
+], function (
+  config,
+  graphicsUtils,
+
+  lang,
+  Deferred,
+  all,
+  topic,
+  projectPopupTemplate,
+
+  TooltipDialog,
+  popup,
+
+  FeatureLayer,
+  Query
+) {
+    return {
+      // scaleTrigger: int
+      // summary:
+      //      the trigger value to where the centroids explode to featureSelected
+      scaleTrigger:  8,
+
+      // override: bool
+      // summary:
+      //      override the trigger
+      override: false,
+
+      // where:  String
+      // summary:
+      //      the expression used to hide the project being looked at or only show centroids
+      //      for a certain collection of projects
+      where: '',
+
+      // centroidLayer: FeatureLayer
+      // summary:
+      //      the centroid feature layer
+      centroidLayer: null,
+
+      // explodedLayer: FeatureLayer
+      // summary:
+      //      the exploded projects
+      explodedLayer: {},
+
+      // enabled: boolean
+      // summary:
+      //      whether or not to care about this stuff
+      enabled: true,
+
+      startup: function () {
+          // summary:
+          //      get everyting started and setup
+          console.log('app.centroidController::startup', arguments);
+
+          topic.subscribe(config.topics.map.extentChanged, lang.hitch(this, 'onExtentChanged'));
+          topic.subscribe(config.topics.map.toggleCentroids, lang.hitch(this, 'updateOverride'));
+
+          this.dialog = new TooltipDialog({
+              id: 'tooltipDialog',
+              style: 'position: absolute; max-width: 250px; z-index:100',
+              onMouseLeave: lang.hitch(this, lang.partial(this._showPopupForProject, false))
```

Comment:
Sounds good to me. I'll go back to `mouse-out`, i'm not going to change the size since I think Aaron is going to change this anyway, i'll turn the cursor into a pointer and add a click event to go to the single project view.